spi/common/core-spi/src/main/java/org/eclipse/edc/spi/types/domain/HttpDataAddress.java

```diff
@@ -222,7 +224,7 @@ public Builder nonChunkedTransfer(boolean nonChunkedTransfer) {
         }
 
         public Builder copyFrom(DataAddress other) {
-            other.getProperties().forEach(this::property);
+            Optional.ofNullable(other).map(DataAddress::getProperties).orElse(emptyMap()).forEach(this::property);
```

Comment:
nit: static import for `ofNullable`

Reply:
I'm not a big fan of static imports when they hide details, in this case `ofNullable` does not explain well what's about to do. I would keep it as it is if that's not a problem

Reply:
To my, it is pretty self-explanatory what's going on. Also, with modern IDEs it you'll get a hint anyways. 
But ok, I'm not that hung up on it :)